tools/build.mjs

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+console.log("start");
+const dev = process.argv[2];
+import esbuild from "esbuild";
+
+const plugins = [
+  {
+    name: "watch-plugin",
+    setup(build) {
+      build.onEnd((result) => {
+        // for VS Code task tracking
+        console.log("start");
+        console.log("end");
+      });
+    },
+  },
+];
+
+const buildOptions = {
+  entryPoints: {
+    "./client/dist/node/extension": "./client/src/node/extension.ts",
+    "./server/dist/node/server": "./server/src/node/server.ts",
+  },
+  bundle: true,
+  outdir: ".",
+  platform: "node",
+  external: ["vscode"],
+  loader: {
+    ".properties": "text",
+  },
+  sourcemap: !!dev,
+  minify: !dev,
+  plugins: dev ? plugins : [],
+};
+
+const ctx = await esbuild.context(buildOptions);
+if (dev) {
+  await ctx.watch();
```

Comment:
I updated the build file using the upgrade guide, here: https://github.com/evanw/esbuild/releases/tag/v0.17.0

There's still _some_ issue (although I'm guessing this build will pass). It appears `ctx.watch` does not wait/hang. Instead, it exits immediately. If anyone has capacity to examine this further, that'd be helpful. Otherwise, I'll likely pick it up next week.

Reply:
These issues are now resolved